nestkernel/nest.cpp

```diff
@@ -273,26 +295,33 @@ connect_arrays( long* sources,
         // Store the key-value pair of each parameter in the Dictionary.
         for ( auto& param_pointer_pair : param_pointers )
         {
+          // Increment the pointer to the parameter value.
+          auto* param = param_pointer_pair.second + index_counter;
+
           // Receptor type must be an integer.
           if ( param_pointer_pair.first == names::receptor_type )
           {
-            const auto int_cast_rtype = static_cast< size_t >( *param_pointer_pair.second );
-            if ( int_cast_rtype != *param_pointer_pair.second )
+            const auto int_cast_rtype = static_cast< size_t >( *param );
+
+            if ( std::abs( *param - int_cast_rtype ) > 1e-20 ) // To avoid rounding errors
```

Comment:
The IEEE 754 float64 ("double precision") format can represent integers from −2^53 to 2^53 (−9007199254740992 to 9007199254740992) exactly. Above those values, rounding to nearest multiples of 2, 4, etc. occurs. The integer type is here defined as ``size_t`` but we are also adding an ``int`. Do you know if the result (i.e. ``*param``) is always 2^32-1 or less? In that case, this test can be made exact. If the values of ``*param`` can be higher, this test will trigger even though the user supplied integer numbers (because of the loss of precision).

Reply:
Integer dictionary entries are `long`, so I would cast to `long` here instead of `size_t`. For 64-bit `long` this still does not solve the problem @clinssen referred to. In practice, receptor numbers will always be much smaller than 2^32. To be totally safe, we could have a double test: throw the exception if the value is either `> 1L>>31 or std::abs(...) > 0`. We can compare directly to 0 here instead of 1e-20, since for proper integers we will have exact zero here. And maybe change the variable name to `rtype_as_long`.

Reply:
@clinssen @heplesser Just so I understand, I should change to
```
const auto rtype_as_long = static_cast< long >( *param );

if ( std::abs( *param - rtype_as_long ) > 1L>>31 or std::abs( *param - rtype_as_long ) > 0 )
```

Reply:
```
if ( *param > 1L << 31 or std::abs( *param - rtype_as_long ) > 0 )
```

Reply:
I have updated the `if`, please have another look.